src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -283,5 +283,44 @@ private static Conversion ToConversion(OverloadResolutionResult<MethodSymbol> re
 
             return new Conversion(ConversionKind.MethodGroup, method, methodGroup.IsExtensionMethodGroup);
         }
+
+        public override Conversion GetStackAllocConversion(BoundStackAllocArrayCreation sourceExpression, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            Debug.Assert((object)sourceExpression.Type == null);
+            Debug.Assert(sourceExpression.ElementType != null);
+
+            var pointerConversion = default(Conversion);
+
+            if (sourceExpression.Syntax.IsVariableDeclarationInitialization())
+            {
+                var sourceAsPointer = new PointerTypeSymbol(sourceExpression.ElementType);
+                pointerConversion = ClassifyImplicitConversionFromType(sourceAsPointer, destination, ref useSiteDiagnostics);
+            }
+
+            if (pointerConversion.IsValid)
+            {
+                // Report unsafe errors
+                _binder.ReportUnsafeIfNotAllowed(sourceExpression.Syntax.Location, ref useSiteDiagnostics);
+
+                return Conversion.MakeStackAllocToPointerType(pointerConversion);
```

Comment:
Can this `pointerConversion.IsValid` case move into the `if (IsVariableDeclarationInitialization)` case above? (Otherwise `pointerConversion` will be invalid.) The `pointerConversion` local is only needed inside that `if`.